aop/src/main/java/io/micronaut/aop/internal/intercepted/KotlinInterceptedMethodImpl.java

```diff
@@ -110,11 +111,11 @@ public CompletableFuture<Object> interceptResultAsCompletionStage() {
             );
         }
         @SuppressWarnings("unchecked")
-        CompletableFutureContinuation completableFutureContinuation = new CompletableFutureContinuation((Continuation<Object>) continuation);
+        var completableFutureContinuation = new CompletableFutureContinuation((Continuation<Object>) continuation);
         replaceContinuation.accept(completableFutureContinuation);
         Object result = context.proceed();
         replaceContinuation.accept(continuation);
-        if (result != KotlinUtils.COROUTINE_SUSPENDED) {
+        if (result != null && result != KotlinUtils.COROUTINE_SUSPENDED) {
```

Comment:
Why did you add a null check?

Reply:
Because `InvocationContext.proceed()` method with annotaion `@Nullable`

Reply:
I don't think that is related, lets see if tests are going to pass

Reply:
I can revert it

Reply:
ok, reverted